pkgs/by-name/al/alist/package.nix

```
@@ -0,0 +1,83 @@
+{
+  lib,
+  alist,
+  buildGoModule,
+  fetchFromGitHub,
+  fetchurl,
+  fuse,
+  go,
+  stdenv,
+  installShellFiles,
+  testers,
+}:
+buildGoModule rec {
+  pname = "alist";
+  version = "3.38.0";
+
+  src = fetchFromGitHub {
+    owner = "alist-org";
+    repo = "alist";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-1nqveTEYrN9kYAOMTbLPw0MFNNNbFOqej+d+aTLAZ1o=";
+  };
+  web = fetchurl {
+    url = "https://github.com/alist-org/alist-web/releases/download/${version}/dist.tar.gz";
+    hash = "sha256-jHbWhjvHfgtdocuuELbOwrMzL8tOQfBVdH9MxasEwGo=";
+  };
+
+  CGO_ENABLED = 1;
+
+  vendorHash = "sha256-qSbvb2/y5rdS/OCutNEcRDUQBCAgNudux8XDnY9TRSo=";
+
+  buildInputs = [ fuse ];
+
+  tags = [ "jsoniter" ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X \"github.com/alist-org/alist/v3/internal/conf.GitAuthor=Xhofe <i@nn.ci>\""
+    # time should not be used as the input for nix, use "Nix" as a placeholder instead.
+    "-X github.com/alist-org/alist/v3/internal/conf.BuiltAt=Nix"
+    "-X github.com/alist-org/alist/v3/internal/conf.GoVersion=${go.version}"
+    "-X github.com/alist-org/alist/v3/internal/conf.GitCommit=${version}"
```

```suggestion
    "-X github.com/alist-org/alist/v3/internal/conf.GitCommit=${src.rev}"
```

IMO, `src.rev` is slightly better than `version`.

There are also [examples][1] using the commit hash.

[1]: https://github.com/NixOS/nixpkgs/blob/e08d91122e2a42091232238db136fc3c12112613/pkgs/tools/admin/bom/default.nix#L41